gravitee-apim-rest-api/gravitee-apim-rest-api-portal/gravitee-apim-rest-api-portal-rest/src/main/java/io/gravitee/rest/api/portal/rest/mapper/PortalNavigationItemMapper.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright © 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.rest.api.portal.rest.mapper;
+
+import io.gravitee.apim.core.portal_page.model.PortalArea;
+import io.gravitee.apim.core.portal_page.model.PortalNavigationFolder;
+import io.gravitee.apim.core.portal_page.model.PortalNavigationItem;
+import io.gravitee.apim.core.portal_page.model.PortalNavigationItemId;
+import io.gravitee.apim.core.portal_page.model.PortalNavigationLink;
+import io.gravitee.apim.core.portal_page.model.PortalNavigationPage;
+import io.gravitee.apim.core.portal_page.model.PortalPageContentId;
+import jakarta.annotation.Nonnull;
+import jakarta.annotation.Nullable;
+import java.util.List;
+import org.mapstruct.Mapper;
+import org.mapstruct.factory.Mappers;
+
+@Mapper
+public interface PortalNavigationItemMapper {
+    PortalNavigationItemMapper INSTANCE = Mappers.getMapper(PortalNavigationItemMapper.class);
+
+    PortalArea map(io.gravitee.rest.api.portal.rest.model.PortalArea area);
+
+    default List<io.gravitee.rest.api.portal.rest.model.PortalNavigationItem> map(@Nonnull List<PortalNavigationItem> items) {
+        return items.stream().map(this::getBasePortalNavigationItem).toList();
+    }
+
+    private io.gravitee.rest.api.portal.rest.model.PortalNavigationItem getBasePortalNavigationItem(PortalNavigationItem item) {
+        var baseItem = switch (item) {
+            case PortalNavigationFolder folder -> map(folder);
+            case PortalNavigationLink link -> map(link);
+            case PortalNavigationPage page -> map(page);
+        };
+        var wrappedItem = new io.gravitee.rest.api.portal.rest.model.PortalNavigationItem();
+        wrappedItem.setActualInstance(baseItem);
+        return wrappedItem;
+    }
```

Comment:
Tiny simplification to have the same pattern as in [PortalNavigationItemsMapper](https://github.com/gravitee-io/gravitee-api-management/blob/7f646b5cb4fa3d9d8f7a30b25584b838bd0960a7/gravitee-apim-rest-api/gravitee-apim-rest-api-management-v2/gravitee-apim-rest-api-management-v2-rest/src/main/java/io/gravitee/rest/api/management/v2/rest/mapper/PortalNavigationItemsMapper.java#L32)

```suggestion

    private io.gravitee.rest.api.portal.rest.model.PortalNavigationItem getBasePortalNavigationItem(PortalNavigationItem item) {
        return switch (item) {
            case PortalNavigationFolder folder -> new io.gravitee.rest.api.portal.rest.model.PortalNavigationItem(map(folder));
            case PortalNavigationLink link -> new io.gravitee.rest.api.portal.rest.model.PortalNavigationItem(map(link));
            case PortalNavigationPage page -> new io.gravitee.rest.api.portal.rest.model.PortalNavigationItem(map(page));
        };
    }

Reply:
As discussed we can keep it as is